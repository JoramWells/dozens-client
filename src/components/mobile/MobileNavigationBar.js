import React from "react";

function MobileNavigationBar() {
  return (
    <nav className="flex flex-row p-2 justify-between shadow-sm">
      <div>1</div>
      <div>2</div>
      <div>3</div>
    </nav>
  );
}

export default MobileNavigationBar;
