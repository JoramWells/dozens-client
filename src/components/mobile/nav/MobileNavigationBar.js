import React, { useCallback, useState } from "react";
import {
  ArrowNarrowLeftIcon,
  HomeIcon,
  SearchIcon,
  XIcon,
} from "@heroicons/react/solid";
import { Link } from "react-router-dom";
import NavDropdown from "../NavDropdown";

function MobileNavigationBar() {
  const [visible, setVisible] = useState("hidden");
  const [diVisible, setDiVisible] = useState("visible");

  const showNav = useCallback(() => {
    setVisible("visible");
    setDiVisible("hidden");
  }, [setVisible, setDiVisible]);

  const hideNav = useCallback(() => {
    setVisible("hidden");
    setDiVisible("visible");
  }, [setDiVisible, setVisible]);

  return (
    <>
      {/* Search nav */}
      <nav
        style={{ visibility: visible, transition: "350ms" }}
        className="fixed top-0 w-full shadow-md p-2 bg-white z-0"
      >
        <div className="flex flex-row justify-center content-center items-center space-x-4">
          <div onClick={() => hideNav()}>
            <ArrowNarrowLeftIcon className="h-5 text-gray-700" />
          </div>
          <div className=" bg-gray-100 flex flex-row items-center content-center rounded-full">
            <SearchIcon className="h-5 m-2 text-gray-500" />
            <input placeholder="Search..." className=" bg-transparent p-1" />
            <XIcon className="h-5 m-2 text-gray-500" />
          </div>
        </div>
      </nav>

      {/* Home nav */}
      <nav
        className="flex flex-row p-3 justify-between items-center shadow-lg fixed w-full z-10 bg-black"
        style={{ visibility: diVisible, transition: "350ms" }}
        data-testid="mobile_nav"
      >
        <div className="flex flex-row space-x-2">
          <Link to="/">
            <HomeIcon className="h-5 text-white" />
          </Link>
        </div>
        <div className="flex flex-row space-x-4 items-center">
          <SearchIcon className="h-5 text-white" onClick={showNav} />

          <NavDropdown />
        </div>
      </nav>
    </>
  );
}

export default MobileNavigationBar;
