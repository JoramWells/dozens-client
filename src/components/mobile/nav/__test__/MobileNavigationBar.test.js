import React from "react";
import ReactDOM from "react-dom";
import MobileNavigationBar from "../MobileNavigationBar";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { BrowserRouter, Switch, Route } from "react-router-dom";

afterEach(cleanup);
it("renders without crashing", () => {
  const nav = document.createElement("div");
  ReactDOM.render(
    <BrowserRouter>
      <Switch>
        <Route path="/" component={MobileNavigationBar} />
      </Switch>
    </BrowserRouter>,
    nav
  );
});

it("renders Navbar correctly", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Switch>
        <Route path="/" component={MobileNavigationBar} />
      </Switch>
    </BrowserRouter>
  );
  expect(getByTestId("mobile_nav")).toBeInTheDocument();
});
// it("renders Navbar correctly", () => {
//     const { getByTestId } = render(<MobileNavigationBar />);
//     expect(getByTestId("mobile_nav")).toHaveTextContent("bram");
//   });

// it("matches snapshot",()=>{
//     const tree = renderer.create(<MobileNavigationBar />).toJSON()
//     expect(tree).toMatchSnapshot()
// })
