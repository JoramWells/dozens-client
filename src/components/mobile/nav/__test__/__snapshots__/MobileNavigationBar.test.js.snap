// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <nav
    className="fixed top-0 w-full shadow-md p-2 bg-yellow-400 z-0"
    style={
      Object {
        "transition": "350ms",
        "visibility": "hidden",
      }
    }
  >
    <div
      className="flex flex-row justify-center content-center items-center space-x-4"
    >
      <div
        onClick={[Function]}
      >
        <svg
          className="h-5 text-gray-700"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M7.707 14.707a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414l4-4a1 1 0 011.414 1.414L5.414 9H17a1 1 0 110 2H5.414l2.293 2.293a1 1 0 010 1.414z"
            fillRule="evenodd"
          />
        </svg>
      </div>
      <div
        className=" bg-yellow-200 flex flex-row items-center content-center rounded-full"
      >
        <svg
          className="h-5 m-1 text-gray-500"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
            fillRule="evenodd"
          />
        </svg>
        <input
          className=" bg-transparent p-1"
          placeholder="Search..."
        />
        <svg
          className="h-5 m-1 text-gray-500"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
            fillRule="evenodd"
          />
        </svg>
      </div>
    </div>
  </nav>,
  <nav
    className="flex flex-row p-3 justify-between shadow-md bg-yellow-400"
    data-testid="mobile_nav"
    style={
      Object {
        "transition": "350ms",
        "visibility": "visible",
      }
    }
  >
    <div
      className="flex flex-row space-x-2"
    >
      home
      <svg
        className="h-5 text-gray-700"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.707 2.293a1 1 0 00-1.414 0l-7 7a1 1 0 001.414 1.414L4 10.414V17a1 1 0 001 1h2a1 1 0 001-1v-2a1 1 0 011-1h2a1 1 0 011 1v2a1 1 0 001 1h2a1 1 0 001-1v-6.586l.293.293a1 1 0 001.414-1.414l-7-7z"
        />
      </svg>
    </div>
    <div>
      <svg
        className="h-5 text-gray-700"
        fill="currentColor"
        onClick={[Function]}
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div>
      <svg
        className="h-5 text-gray-700"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z"
          fillRule="evenodd"
        />
      </svg>
    </div>
  </nav>,
]
`;
